### PR TITLE
small paths code rearrangement

### PR DESCRIPTION
**Pull request by ryanfkeepers**
small cleanup in paths, primarily splitting files
so that file contents are more clearly owned, which should be a little better for readability and code placement.

Also renames `ServicePrefix` to `BuildPrefix` in
anticipation of multi-service prefixes.

no logic changes, just movement/renaming.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #3993

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by ashmrtn — APPROVED** [review]

File: src/pkg/path/drive.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package path
 
 import "github.com/alcionai/clues"
 
+// TODO: Move this into m365/collection/drive
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/path/drive.go:5]: okedoke, removed the TODO for now.

File: src/pkg/path/builder.go [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+// Resource-specific paths allow access to more information like segments in the
+// path. Builders that are turned into resource paths later on do not need to
+// manually add prefixes for items that normally appear in the data layer (ex.
+// tenant ID, service, user ID, etc).
+type Builder struct {
+	// Unescaped version of elements.
+	elements Elements
+}
+
+// Append creates a copy of this Builder and adds the given elements them to the
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/path/builder.go:30]: done!

File: src/pkg/path/builder.go [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+			service.String(),
+			user,
+			category.String()),
+		service:  service,
+		category: category,
+		hasItem:  isItem,
+	}, nil
+}
+
+func (pb Builder) ToDataLayerExchangePathForCategory(
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/path/builder.go:326]: done! https://github.com/alcionai/corso/issues/4025

File: src/pkg/path/builder.go [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+// String returns a string that contains all path elements joined together.
+// Elements of the path that need escaping are escaped.
+// The result is not concealed, and is not suitable for logging or structured
+// errors.
+func (pb Builder) String() string {
+	return pb.elements.String()
+}
+
+// PlainString returns an unescaped, unmodified string of the builder.
+// The result is not concealed, and is not suitable for logging or structured
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/path/builder.go:375]: Not necessarily.  Check out the difference between Elements' String and PlainString methods: https://github.com/alcionai/corso/blob/main/src/pkg/path/elements.go#L88.  The primary difference is that `String` will escape each element, PlainString won't.

**Pull request merged**